### PR TITLE
BUGFIX: Prevent double encoding in remaining form elements

### PR DESCRIPTION
**Pull request by gerhard-boden**
Use `translate` ViewHelper to prevent double encoding for the `text` and `label` property.

### CODE REVIEW

**Comment by gerhard-boden** [comment]
The idea for this fix came from @bwaidelich, thanks!

When looking at https://github.com/neos/form/commit/9d3e4afad25ed4c26e768b203b90a489b7f03ef2#diff-c35d5c121ce57b6977905879ec9c123a I think there are a couple of fields that need a similar change.

**Comment by aertmann** [comment]
yep definitely seems like there are more of these issues that need to be fixed /cc @kdambekalns

**Comment by gerhard-boden** [comment]
Took care of it, fixed the remaining form elements that had double encoding issues. Also used the ``TranslateElementPropertyViewHelper`` from @bwaidelich where it made sense (translation id is the same as the property name). Makes the markup much more readable.

To speed up testing use the attached `contact-form.yaml`` (had to rename it to .txt):
[contact-form.txt](https://github.com/neos/form/files/367548/contact-form.txt)
It contains all form elements with special some special characters.
Replace it with the standard form configuration in ``Packages/Sites/Neos.Demo/Resources/Private/Form/contact-form.yaml``

All the special characters are encoded correctly now within the form elements, as far is can tell from the source code.

**Comment by gerhard-boden** [comment]
And another question about file permissions: They were changed in my setup and are therefore also different  in the commited files. Is that something to avoid or are file permissions reset at some point?

**Comment by bwaidelich** [comment]
Great work, thank you! (didn't test yet)

**Comment by kdambekalns** [comment]
Looks good, thanks! And the permissions should be 644 rather than 755, those files need not be executable.

**Comment by gerhard-boden** [comment]
Will change the permission. I always run `core:setfilepermissions` after a fresh installation. Maybe that's causing the issue?

**Comment by kdambekalns** [comment]
But that should not make those files executable!?

**Comment by gerhard-boden** [comment]
Will verify this in the evening with a fresh installation. Hopefully the wrong file permssions are  not caused by the script.

**Comment by gerhard-boden** [comment]
Ran the file permission script again, the file permissions look good now,
https://github.com/neos/form/pull/18/commits/9f9ddb45eb6223b01d71edfdad8b018fbf67a98a

**Comment by bwaidelich** [comment]
FYI: I'll try too look into this one asap

File: Resources/Private/Form/Password.html [hunk 1]
```diff
@@ -1,7 +1,8 @@
+{namespace form=Neos\Form\ViewHelpers}
 <f:layout name="Neos.Form:Field" />
 <f:section name="field">
 	<f:form.password property="{element.identifier}" id="{element.uniqueIdentifier}" class="{element.properties.elementClassAttribute}" errorClass="{element.properties.elementErrorClassAttribute}" />
 	<f:if condition="{element.properties.passwordDescription}">
-		<span class="help-block">{element.properties.passwordDescription -> f:translate(id: 'forms.elements.{element.identifier}.passwordDescription', package: '{element.renderingOptions.translationPackage}')}</span>
+		<span class="help-block">{element -> form:translateElementProperty(property: 'passwordDescription')}</span>
```
Comment (kdambekalns) [inline Resources/Private/Form/Password.html]: Does `translateElementProperty()` automagically use `element.renderingOptions.translationPackage`?

Comment (bwaidelich) [inline Resources/Private/Form/Password.html]: Yes it does, see https://github.com/neos/form/blob/master/Classes/ViewHelpers/TranslateElementPropertyViewHelper.php#L68

**Comment by bwaidelich** [comment]
Oh wow, thanks for taking over here. I had some "unplanned trouble" so I couldn't take care yet

File: Resources/Private/Form/Password.html (hunk 1 above)
Comment (kdambekalns) [inline Resources/Private/Form/Password.html]: Great! Thanks for solving my laziness. 😄

File: Resources/Private/Form/Partials/Form/Navigation.html [hunk 2]
```diff
@@ -16,7 +16,7 @@
 					<f:form.button name="__currentPage" value="{form.pages -> f:count()}" class="btn btn-primary">
 						<f:if condition="{form.renderingOptions.submitButtonLabel}">
 							<f:then>
-								{form.renderingOptions.submitButtonLabel -> f:translate(package: '{form.renderingOptions.translationPackage}')}
+								{f:translate(id: 'forms.navigation.{form.identifier}.submit', package: '{form.renderingOptions.translationPackage}', value: form.renderingOptions.submitButtonLabel)}
```
Comment (bwaidelich) [inline Resources/Private/Form/Partials/Form/Navigation.html]: Why is the form identifier part of the l10n id now? in the other labels that's not the case, for example `forms.navigation.nextPage`

**Comment by bwaidelich** [comment]
..as follow-up to my comment above: the form identifier is now only part of the `forms.navigation.{form.identifier}.submit` label. IMO we should avoid that.
When you want custom labels for a form, just change the `translationPackage` Setting of that form. WDYT?

File: Resources/Private/Form/StaticText.html [hunk 3]
```diff
@@ -1,6 +1,7 @@
+{namespace form=TYPO3\Form\ViewHelpers}
```
Comment (bwaidelich) [inline Resources/Private/Form/StaticText.html]: OOps, "TYPO3" ;)

**Comment by kdambekalns** [comment]
Oops. :)

BTW, it is really, really easy to click edit in GitHub and just fix those tiny things. Takes about as long as leaving a comment about it. ;)

**Comment by bwaidelich** [comment]
I just fixed it locally, wanted to test if first ;)

**Comment by bwaidelich** [comment]
I also adjusted the submit button translation id as suggested. Merging with previous votes. Thanks for taking care of this and sorry for the delay ;)

File: Resources/Private/Form/StaticText.html (hunk 3 above)
Comment (gerhard-boden) [inline Resources/Private/Form/StaticText.html]: This PR is older than the namespace adjustment 😄 

But yea, I should have updated in the meantime. Will need to do that for my other old PRs as well

**Comment by gerhard-boden** [comment]
@bwaidelich @kdambekalns 
Thanks a lot guys for reviewing and updating this PR!

File: Resources/Private/Form/StaticText.html (hunk 3 above)
Comment (bwaidelich) [inline Resources/Private/Form/StaticText.html]: @gerhard-boden don't worry, it was not easy to spot - I only discovered it by chance ;)

**Pull request merged**